packages/governance/src/ballotBuilder.js

```diff
@@ -21,64 +20,160 @@ const ChoiceMethod = {
   WEIGHT: 'weight',
 };
 
-const buildEqualWeightBallot = (
+/** @type {{
+ *   PARAM_CHANGE: 'param_change',
+ *   ELECTION: 'election',
+ *   SURVEY: 'survey',
+ * }}
+ */
+const ElectionType = {
+  // A parameter is named, and a new value proposed
+  PARAM_CHANGE: 'param_change',
+  // choose one or multiple winners, depending on ChoiceMethod
+  ELECTION: 'election',
+  SURVEY: 'survey',
+};
+
+/** @type {{
+ *   MAJORITY: 'majority',
+ *   NO_QUORUM: 'no_quorum',
+ *   ALL: 'all',
+ * }}
+ */
+const QuorumRule = {
+  MAJORITY: 'majority',
+  NO_QUORUM: 'no_quorum',
+  // The election isn't valid unless all voters cast a ballot
+  ALL: 'all',
+};
+
+const verifyQuestionFormat = (electionType, question) => {
```

Comment:
I don't understand the type of the `question` parameter of `verifyQuestionFormat`. A `Question` is either a `SimpleQuestion` or a `ParamChangeQuestion`. For the first, according to types.js, the property name is `question`, not `text`.

(IIRC, @Chris-Hibbert and I resolved in chat that the type should be revised to agree with the code)

Reply:
For a lot of these checking functions, I'm not always clear whether they're just heuristically trying to spot some errors, in order to give better diagnostics earlier (which is great), vs if they're trying to do a full input validation so that we can count on various invariants if the check passes. Which is this? With a name like "verifySomething" I'd guess that it's the latter. But this code doesn't even check that `question` is frozen or passable. (Nevermind that we expect better tool support for checking coming soon. The code here should be correct with the tools we have now.) A check like

```suggestion
const verifyQuestionFormat = (electionType, question) => {
  assert(passStyleOf(question) === 'copyRecord');
```
takes care of many of these issues. `passStyleOf` memoizes its work, so repeated checks of the same object should be cheap.

If the intention is only heuristic bug finding and diagnostics, I like our existing `looksLikeFoo` convention to make the distinction clear.

Reply:
I renamed to `looksLikeFoo`.

I aspire to formal validation, but I don't think I know the complete set of properties that should be checked.